IM_calculation/Advanced_IM/Models/Spear_3D_Model/run.py

```diff
@@ -0,0 +1,311 @@
+
+    if os.path.exists(im_gravity_recorder):
+        gr_value = float(read_out_file(im_gravity_recorder))
+    else:
+        gr_value = 0
+
+    im_recorder = os.path.join(
+        direction_dir, os.path.join(recorder_name, im_recorder_fname)
+    )
+
+    with open(im_recorder) as f_im_recorder:
+        im_records_list_tmp = [float(line.split()[1]) for line in f_im_recorder]
+    im_records_list = []
+    for im_record in im_records_list_tmp[:-1]:
+        im_record = im_record - gr_value
+        im_records_list.append(im_record)
+
+    return im_records_list
+
+
+def calculate_norm(im_name, output_dir, print_header=True):
+    # get 000 and 090 dir
+    dir_000 = os.path.join(output_dir, "000")
+    dir_090 = os.path.join(output_dir, "090")
+    component = "norm"
+
+    im_csv_fname = os.path.join(output_dir, im_name + "_" + component + ".csv")
+
+    df_combined = {}
+
+    value_dict = {}
+
+    # read data from recordings
+    for recorder_name in ["disp", "drift", "accl"]:
+        # find recorder in 000 first
+        # read all 2nd coloumn
+        recorder_dir = os.path.join(dir_000, recorder_name)
+        im_recorder_glob = os.path.join(recorder_dir, "*.out")
+        im_recorders = glob.glob(im_recorder_glob)
+
+        for im_recorder in im_recorders:
+            im_recorder_fname = os.path.basename(im_recorder)
+            im_recorder_name = os.path.splitext(im_recorder_fname)[0]
+            im_record_list = {}
+
+            im_record_list["000"] = read_recorder_all(
+                dir_000, recorder_name, im_recorder_fname
+            )
+            im_record_list["090"] = read_recorder_all(
+                dir_090, recorder_name, im_recorder_fname
+            )
+            im_record_list["norm"] = []
+            if len(im_record_list["000"]) != len(im_record_list["090"]):
+                raise (
+                    ValueError(
+                        f"length of 000 and 090 does not match for {im_recorder_fname}"
+                    )
+                )
+            # calulate norm
+            for i in range(0, len(im_record_list["000"])):
+                norm = np.sqrt(
+                    (
+                        np.power(im_record_list["000"][i], 2)
+                        + np.power(im_record_list["090"][i], 2)
+                    )
+                )
+                im_record_list["norm"].append(norm)
+
+            im_value = max(im_record_list["norm"])
+            # value_dict[recorder_name][im_recorder_name] = im_value
+            value_dict[im_recorder_name] = im_value
+
+    value_dict = {component: value_dict}
+    result_df = pd.DataFrame.from_dict(value_dict, orient="index")
+
+    cols = list(result_df.columns)
+    cols.sort()
+
+    # test if file exist, if exist, no header
+    if os.path.isfile(im_csv_fname):
+        print_header = False
+
+    result_df.to_csv(im_csv_fname, mode="a", header=print_header, columns=cols)
+
+
+def calculate_geom(output_dir, im_name):

```

Comment:
Add a docstring to say this func takes two files in and generates the geom from them?